### PR TITLE
Data Sources For Power Systems: workspace/s & datacenter/s

### PR DESCRIPTION
**Pull request by michaelkad**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates OR Closes #0000

Output from acceptance testing:

```
=== RUN   TestAccIBMPIWorkspaceDataSourceBasic
--- PASS: TestAccIBMPIWorkspaceDataSourceBasic (16.31s)
PASS

=== RUN   TestAccIBMPIWorkspacesDataSourceBasic
--- PASS: TestAccIBMPIWorkspacesDataSourceBasic (37.10s)
PASS

=== RUN   TestAccIBMPIDatacentersDataSourceBasic
--- PASS: TestAccIBMPIDatacentersDataSourceBasic (8.51s)
PASS

=== RUN   TestAccIBMPIDatacenterDataSourceBasic
--- PASS: TestAccIBMPIDatacenterDataSourceBasic (9.04s)
PASS

```

### CODE REVIEW

File: website/docs/d/pi_datacenter.html.markdown [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+- `pi_datacenter_location` - (Map) Datacenter location.
+
+    Nested schema for `pi_datacenter_location`:
+  - `region` - (String) The Datacenter location region zone.
+  - `type` - (String) The Datacenter location region type.
+  - `url`- (String) The Datacenter location region url.
+- `pi_datacenter_status` - (String) The Datacenter status, `ACTIVE`,`MAINTENANCE` or `DOWN`.
+- `pi_datacenter_type` - (String) The Datacenter type, `Public Cloud` or `Private Cloud`.
+
+## Notes
```
Comment (yussufsh) [inline website/docs/d/pi_datacenter.html.markdown:50]: Can we move this `notes` in `Example usage` section? eg: https://github.com/IBM-Cloud/terraform-provider-ibm/blob/master/website/docs/d/pi_image.html.markdown

File: ibm/service/power/ibm_pi_constants.go [hunk 2]
```diff
@@ -5,6 +5,7 @@ import "time"
 const (
 	// used by all
 	Arg_CloudInstanceID = "pi_cloud_instance_id"
+	Arg_DatacenterZone  = "pi_datacenter_zone"
```
Comment (yussufsh) [inline ibm/service/power/ibm_pi_constants.go:8]: Move this to line 163.

File: ibm/service/power/data_source_ibm_pi_datacenter.go [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+	DatacenterRegion = "region"
+	DatacenterType   = "type"
+	DatacenterUrl    = "url"
+)
+
+func DataSourceIBMPIDatacenter() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataSourceIBMPIDatacenterRead,
+		Schema: map[string]*schema.Schema{
+			Arg_DatacenterZone: {
```
Comment (yussufsh) [inline ibm/service/power/data_source_ibm_pi_datacenter.go:28]: If it is DC region. Why are we calling it a `zone` in Terraform?

File: ibm/service/power/data_source_ibm_pi_datacenter.go [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+func dataSourceIBMPIDatacenterRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	// session
+	sess, err := meta.(conns.ClientSession).IBMPISession()
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	datacenterRegion := sess.Options.Zone
+	if region, ok := d.GetOk(Arg_DatacenterZone); ok {
+		datacenterRegion = region.(string)
```
Comment (yussufsh) [inline ibm/service/power/data_source_ibm_pi_datacenter.go:73]: Zone=Region. It is confusing 👎

File: ibm/service/power/data_source_ibm_pi_datacenters.go [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+		},
+	}
+}
+func dataSourceIBMPIDatacentersRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	// session
+	sess, err := meta.(conns.ClientSession).IBMPISession()
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	client := instance.NewIBMPIDatacenterClient(ctx, sess, "")
```
Comment (yussufsh) [inline ibm/service/power/data_source_ibm_pi_datacenters.go:70]: @michaelkad Does it not need a cloud instance ID? How is the auth/CRN created then?

File: ibm/service/power/data_source_ibm_pi_datacenter.go [hunk 6]
```diff
@@ -0,0 +1,94 @@
+// Copyright IBM Corp. 2017, 2021 All Rights Reserved.
```
Comment (yussufsh) [inline ibm/service/power/data_source_ibm_pi_datacenter.go:1]:
```suggestion
// Copyright IBM Corp. 2023 All Rights Reserved.
```

Comment (yussufsh) [reply ibm/service/power/data_source_ibm_pi_datacenter.go:1]: Add this to every new file.

File: ibm/service/power/data_source_ibm_pi_datacenters.go (hunk 5 above)
Comment (yussufsh) [reply ibm/service/power/data_source_ibm_pi_datacenters.go:70]: When I ran the example, the api throws 404.

File: website/docs/d/pi_workspaces.html.markdown [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+
+Review the argument references that you can specify for your data source.
+
+- `pi_cloud_instance_id` - (Required, String) Cloud Instance ID of a PCloud Instance.
+
+## Attribute reference
+
+In addition to all argument reference list, you can access the following attribute references after your data source is created.
+
+- `workspaces` - List of of all Datacenters.
```
Comment (yussufsh) [inline website/docs/d/pi_workspaces.html.markdown:32]:
```suggestion
- `workspaces` - List of all Workspaces.
```

File: website/docs/d/pi_datacenter.html.markdown [hunk 8]
```diff
@@ -0,0 +1,64 @@
+---
+
+subcategory: "Power Systems"
+layout: "ibm"
+page_title: "IBM: pi_datacenter"
+description: |-
+  Manages a Datacenter in the Power Virtual Server cloud.
+---
+
+# ibm_pi_Datacenter
```
Comment (yussufsh) [inline website/docs/d/pi_datacenter.html.markdown:10]:
```suggestion
# ibm_pi_datacenter
```

File: website/docs/d/pi_datacenters.html.markdown [hunk 9]
```diff
@@ -0,0 +1,61 @@
+---
+
+subcategory: "Power Systems"
+layout: "ibm"
+page_title: "IBM: pi_datacenters"
+description: |-
+  Manages Datacenters in the Power Virtual Server cloud.
+---
+
+# ibm_pi_Datacenter
```
Comment (yussufsh) [inline website/docs/d/pi_datacenters.html.markdown:10]:
```suggestion
# ibm_pi_datacenters
```

File: website/docs/d/pi_workspaces.html.markdown [hunk 10]
```diff
@@ -0,0 +1,73 @@
+---
+
+subcategory: "Power Systems"
+layout: "ibm"
+page_title: "IBM: pi_workspaces"
+description: |-
+  Manages  workspaces in the Power Virtual Server cloud.
+---
+
+# ibm_pi_workspace
```
Comment (yussufsh) [inline website/docs/d/pi_workspaces.html.markdown:10]:
```suggestion
# ibm_pi_workspaces
```

File: website/docs/d/pi_workspaces.html.markdown [hunk 11]
```diff
@@ -0,0 +1,73 @@
...
+
+Retrieve information about  Power Systems workspaces.
+
+## Example usage
+
+```terraform
+data "ibm_pi_workspace" "workspace" {
```
Comment (yussufsh) [inline website/docs/d/pi_workspaces.html.markdown:17]:
```suggestion
data "ibm_pi_workspaces" "workspace" {
```

**Review by yussufsh — CHANGES_REQUESTED** [review]

File: website/docs/d/pi_datacenter.html.markdown (hunk 1 above)
Comment (michaelkad) [reply website/docs/d/pi_datacenter.html.markdown]: Moved `notes` in all of them

File: ibm/service/power/ibm_pi_constants.go (hunk 2 above)
Comment (michaelkad) [reply ibm/service/power/ibm_pi_constants.go]: Moved to `163`

File: ibm/service/power/data_source_ibm_pi_datacenter.go (hunk 3 above)
Comment (michaelkad) [reply ibm/service/power/data_source_ibm_pi_datacenter.go:28]: it is a zone — I have updated [data_source_ibm_pi_datacenter.go](https://github.com/IBM-Cloud/terraform-provider-ibm/pull/4874/files/92e3d44a4d96e3847a045d38adcba43d0af19f96#diff-139042b4df2bbc9be53090ac45feeab4c02ad0f221788a9426b8e8b421b55a6c)
` provider "ibm” {

    region    =   “lon"

    zone      =   "lon04"
  }`

File: website/docs/d/pi_datacenter.html.markdown (hunk 8 above)
Comment (michaelkad) [reply website/docs/d/pi_datacenter.html.markdown]: Updated

File: website/docs/d/pi_datacenters.html.markdown (hunk 9 above)
Comment (michaelkad) [reply website/docs/d/pi_datacenters.html.markdown]: Updated

File: website/docs/d/pi_workspaces.html.markdown (hunk 10 above)
Comment (michaelkad) [reply website/docs/d/pi_workspaces.html.markdown]: Updated

File: website/docs/d/pi_workspaces.html.markdown (hunk 11 above)
Comment (michaelkad) [reply website/docs/d/pi_workspaces.html.markdown]: updated

File: website/docs/d/pi_workspaces.html.markdown (hunk 7 above)
Comment (michaelkad) [reply website/docs/d/pi_workspaces.html.markdown]: updated

File: ibm/service/power/data_source_ibm_pi_datacenter.go (hunk 4 above)
Comment (michaelkad) [reply ibm/service/power/data_source_ibm_pi_datacenter.go]: Update to `datacenterZone` to clear confusion

File: ibm/service/power/data_source_ibm_pi_datacenter.go (hunk 6 above)
Comment (michaelkad) [reply ibm/service/power/data_source_ibm_pi_datacenter.go]: Updated

**Review by yussufsh — APPROVED** [review]
lgtm
@michaelkad Please squash the commits into 1.
cc @hkantare

**Pull request merged**